mxcubeweb/core/server/resource_handler.py

```diff
@@ -0,0 +1,500 @@
+
+            # Prepare data for all required arguments
+            validated_data = {}
+            param_data = self._extract_param_data(http_method)
+
+            # Validate parameters from the request
+            for param_name, param_type in annotations.items():
+                if param_name == "return":  # Skip the return annotation
+                    continue
+
+                try:
+                    if param_data is not None:
+                        validated_data[param_name] = self._validate_param_data(
+                            param_name, param_type, param_data
+                        )
+                except (ValueError, TypeError) as ex:
+                    log.error(str(ex))  # noqa: TRY400
+                    msg = f"Invalid input for '{param_name}'"
+                    return (
+                        jsonify({"error": msg}),
+                        400,
+                    )
+
+            # Call the view function with validated data
+            try:
+                result = view_func(**validated_data)
+            except Exception:
+                msg = "Exception raised when calling view function"
+                log.exception(msg)
+                return jsonify({"error": msg}), 500
+            else:
+                # Handle and serialize the result
+                return self._handle_view_result(result)
+
+        return _view_func
+
+    def _assert_pydantic_arguments(self, export):
+        """
+        Ensures the method referenced in the export uses Pydantic arguments.
+        """
+        obj = next(iter(self._adapter_dict.values()))
+        assert_pydantic_arguments(getattr(obj, export["attr"]))
+
+    def _create_openapi_doc_for_view(self, route, export):
+        """
+        Adds OpenAPI documentation for a route.
+        """
+        # Get the first adapter object, the signature are all the same (same class) so
+        # any will do for documentation purpose
+        http_method = export["method"]
+        obj = next(iter(self._adapter_dict.values()))
+        view_func = getattr(obj, export["attr"])
+        annotations = view_func.__annotations__
+
+        # Add OpenAPI documentation root for route
+        self.openapi_spec.add_openapi_path(
+            self._url_prefix, route, export, http_method, view_func
+        )
+
+        # Loop through annotations add response and arguments to OpenAPI spec for route
+        for param_name, param_type in annotations.items():
+            if param_name == "return":
+                self.openapi_spec.add_openapi_response(
+                    self._url_prefix, route, http_method, param_type
+                )
+                continue
+
+            # If it's a Pydantic model, document
+            if issubclass(param_type, BaseModel):
+                self.openapi_spec.add_openapi_schema(
+                    self._url_prefix, route, http_method, param_type
+                )
+
+    def _extract_param_data(self, http_method) -> any:
```

Comment:
```suggestion
    def _extract_param_data(self, http_method) -> object:
```

I am not sure `any` with lower case even exists, does it?

Reply:
:), I thought so